velox/type/Type.cpp

```diff
@@ -1008,6 +1008,8 @@ std::string DateType::toString(int32_t days) const {
       days);
   TimestampToStringOptions options;
   options.mode = TimestampToStringOptions::Mode::kDateOnly;
+  // Enable zero-padding for year, to ensure compliance with 'YYYY' format.
+  options.zeroPaddingYear = true;
```

Comment:
Please, extend TEST(TypeTest, dateToString)  to cover this change and update PR description to mention it.